grades/api_test.py

```diff
@@ -0,0 +1,232 @@
+"""
+Tests for grades API
+"""
+from datetime import datetime, timedelta
+from unittest.mock import patch, MagicMock
+
+import pytz
+from django.core.exceptions import ImproperlyConfigured
+
+from backends.edxorg import EdxOrgOAuth2
+from courses.factories import CourseRunFactory
+from dashboard.api_edx_cache import CachedEdxUserData, UserCachedRunData
+from dashboard.factories import (
+    CachedCertificateFactory,
+    CachedCurrentGradeFactory,
+    CachedEnrollmentFactory,
+)
+from grades import api
+from grades.models import FinalGrade, FinalGradeStatus
+from micromasters.factories import UserFactory
+from search.base import ESTestCase
+
+
+# pylint: disable=no-self-use, protected-access
+
+
+class GradeAPITests(ESTestCase):
+    """
+    Tests for final grades api
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory.create()
+        cls.user.social_auth.create(
+            provider=EdxOrgOAuth2.name,
+            uid="{}_edx".format(cls.user.username),
+            extra_data={"access_token": "fooooootoken"}
+        )
+
+        cls.course_run1 = CourseRunFactory.create(
+            freeze_grade_date=datetime.now(tz=pytz.UTC)-timedelta(days=1),
+            course__program__financial_aid_availability=True,
+        )
+        cls.course_run2 = CourseRunFactory.create(
+            freeze_grade_date=None,
+            course__program=cls.course_run1.course.program,
+        )
+
+        cls.course_run3 = CourseRunFactory.create(
+            freeze_grade_date=datetime.now(tz=pytz.UTC)+timedelta(days=1),
```

Comment:
ah yes, I wanted to have the assignment to be explicit so it is more readable and we do not need to go and check the default value. No strong opinions, though.